numpy/_core/tests/test_defchararray.py

```diff
@@ -136,121 +137,130 @@ def fail():
 
     def test_less_equal(self):
-        assert_array_equal((self.A <= self.B),
+        (A, B) = (self._create_array_a(), self._create_array_b())
+        assert_array_equal((A <= B),
                            [[True, True, True], [False, True, False]])
 
     def test_greater(self):
-        assert_array_equal((self.A > self.B),
+        (A, B) = (self._create_array_a(), self._create_array_b())
+        assert_array_equal((A > B),
                            [[False, False, False], [True, False, True]])
 
     def test_less(self):
-        assert_array_equal((self.A < self.B),
+        (A, B) = (self._create_array_a(), self._create_array_b())
+        assert_array_equal((A < B),
                            [[True, True, False], [False, True, False]])
 
     def test_type(self):
-        out1 = np.char.equal(self.A, self.B)
+        (A, B) = (self._create_array_a(), self._create_array_b())
+        out1 = np.char.equal(A, B)
         out2 = np.char.equal('a', 'a')
         assert_(isinstance(out1, np.ndarray))
         assert_(isinstance(out2, np.ndarray))
 
+
 class TestComparisonsMixed1(TestComparisons):
     """Ticket #1276"""
 
-    def setup_method(self):
-        TestComparisons.setup_method(self)
-        self.B = np.array(
+    def _create_array_b(self):
+        return np.array(
             [['efg', 'efg', '123  '],
              ['051', 'efgg', 'tuv']], np.str_).view(np.char.chararray)
 
+
 class TestComparisonsMixed2(TestComparisons):
     """Ticket #1276"""
 
-    def setup_method(self):
-        TestComparisons.setup_method(self)
-        self.A = np.array(
+    def _create_array_a(self):
+        return np.array(
             [['abc', 'abcc', '123'],
              ['789', 'abc', 'xyz']], np.str_).view(np.char.chararray)
 
+
 class TestInformation:
-    def setup_method(self):
-        self.A = np.array([[' abc ', ''],
+    def _create_array_a(self):
+        return np.array([[' abc ', ''],
                            ['12345', 'MixedCase'],
                            ['123 \t 345 \0 ', 'UPPER']]) \
                             .view(np.char.chararray)
-        self.B = np.array([[' \u03a3 ', ''],
+
+    def _create_array_b(self):
+        return np.array([[' \u03a3 ', ''],
                            ['12345', 'MixedCase'],
                            ['123 \t 345 \0 ', 'UPPER']]) \
                             .view(np.char.chararray)
-        # Array with longer strings, > MEMCHR_CUT_OFF in code.
-        self.C = (np.array(['ABCDEFGHIJKLMNOPQRSTUVWXYZ',
-                            '01234567890123456789012345'])
-                  .view(np.char.chararray))
 
     def test_len(self):
-        assert_(issubclass(np.char.str_len(self.A).dtype.type, np.integer))
-        assert_array_equal(np.char.str_len(self.A), [[5, 0], [5, 9], [12, 5]])
-        assert_array_equal(np.char.str_len(self.B), [[3, 0], [5, 9], [12, 5]])
+        (A, B) = (self._create_array_a(), self._create_array_b())
```

Comment:
If a test only uses `A`, you could do `(A, _) = self._create_arrays()` to discard `B` without binding it to a variable. By convention in Python a variable named `_` is used for this purpose.

Reply:
also, no need for parentheses here (on both sides)